### PR TITLE
make `search-backend-module-explore` & `search-backend-module-techdocs` configurable from app-config

### PR DESCRIPTION
**Pull request by tudi2d**
- Removing options from backend modules `search-backend-module-explore` & `search-backend-module-techdocs`
- Add respective `config.d.ts` & change effected tests
- Convert `permission-backend` from callback to object in favour of not wanting options in the future

---

- In addition to https://github.com/backstage/backstage/pull/19278
- Works towards https://github.com/backstage/backstage/issues/18390

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search-backend-module-explore/src/alpha.ts [hunk 1]
```diff
@@ -23,66 +23,59 @@ import {
...
+        indexRegistry,
+        tokenManager,
+      }) {
+        const defaultSchedule = {
+          frequency: { minutes: 10 },
+          timeout: { minutes: 15 },
+          initialDelay: { seconds: 3 },
+        };
+
+        const schedule = config.has('search.explore.schedule')
```
Comment (Rugvip) [inline plugins/search-backend-module-explore/src/alpha.ts:64]: I've been moving these into the equivalent of `ToolDocumentCollatorFactory.fromConfig` instead, that way you can still use static configuration in the old system. Thinking that avoids potential confusion about why the config schema doesn't match and doesn't hurt much.

File: plugins/search-backend-module-techdocs/src/collators/DefaultTechDocsCollatorFactory.ts [hunk 2]
```diff
@@ -104,7 +104,18 @@ export class DefaultTechDocsCollatorFactory implements DocumentCollatorFactory {
       config.getOptionalBoolean(
         'techdocs.legacyUseCaseSensitiveTripletPaths',
       ) || false;
-    return new DefaultTechDocsCollatorFactory({ ...options, legacyPathCasing });
+    const locationTemplate = config.getOptionalString(
+      'search.techdocs.collators.locationTemplate',
```
Comment (Rugvip) [inline plugins/search-backend-module-techdocs/src/collators/DefaultTechDocsCollatorFactory.ts:108]: Yeah definitely feeling that `search.collators.techdocs.locationTemplate` would make more sense? 😅

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 😁

File: plugins/search-backend-module-explore/config.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TaskScheduleDefinitionConfig } from '@backstage/backend-tasks';
+
+export interface Config {
+  search?: {
+    explore?: {
```
[the remark this answers is not in the document]
Comment (freben) [reply plugins/search-backend-module-explore/config.d.ts:21]: That's what https://github.com/backstage/backstage/pull/19278 does too

**Review by camilaibs — APPROVED** [review]
LGTM, just left a non-blocking suggestion.

File: plugins/search-backend-module-explore/src/alpha.ts [hunk 4]
```diff
@@ -23,66 +23,59 @@ import {
...
+        tokenManager,
+      }) {
+        const defaultSchedule = {
+          frequency: { minutes: 10 },
+          timeout: { minutes: 15 },
+          initialDelay: { seconds: 3 },
+        };
+
+        const schedule = config.has('search.collators.explore.schedule')
+          ? readTaskScheduleDefinitionFromConfig(
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply plugins/search-backend-module-explore/src/alpha.ts:65]: Perfect, thanks for sharing 🙌🏻

File: plugins/search-backend-module-explore/config.d.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * limitations under the License.
+ */
+
+import { TaskScheduleDefinitionConfig } from '@backstage/backend-tasks';
+
+export interface Config {
+  search?: {
+    collators?: {
+      explore?: {
+        schedule?: TaskScheduleDefinitionConfig;
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply plugins/search-backend-module-explore/config.d.ts:23]: TIL, thanks for sharing about the anchors!

**Pull request merged**